compiler/src/dotty/tools/dotc/tastyreflect/PatternOpsImpl.scala

```diff
@@ -131,10 +131,10 @@ trait PatternOpsImpl extends scala.tasty.reflect.PatternOps with CoreImpl {
 
     object TypeTest extends TypeTestModule {
       def apply(tpt: TypeTree)(implicit ctx: Context): TypeTest =
-        tpd.Typed(untpd.Ident(nme.WILDCARD).withType(tpt.tpe), tpt)
+        withDefaultPos(ctx => tpd.Typed(untpd.Ident(nme.WILDCARD)(ctx.source).withType(tpt.tpe)(ctx), tpt)(ctx))
 
       def copy(original: TypeTest)(tpt: TypeTree)(implicit ctx: Context): TypeTest =
-        tpd.cpy.Typed(original)(untpd.Ident(nme.WILDCARD).withType(tpt.tpe), tpt)
+        tpd.cpy.Typed(original)(untpd.Ident(nme.WILDCARD).withSpan(original.span).withType(tpt.tpe), tpt)
```

Comment:
This seems to be an argument that we should disregard TypeTrees in TastyReflect -- they are not reliable, macros should always depend on types instead of type trees.

Reply:
We need the trees for positions. And this is not a type tree, it is the pattern `_: T` of a `case _: T =>`